### PR TITLE
Create new node ID on each launch & support keepalive with preferred ports

### PR DESCRIPTION
**Pull request by SergiySW**

### CODE REVIEW

File: nano/node/lmdb.cpp [hunk 1]
```diff
@@ -1271,6 +1243,10 @@ void nano::mdb_store::upgrade_v13_to_v14 (nano::transaction const & transaction_
 	}
 
 	logging.logger.always_log (boost::str (boost::format ("Completed confirmation height upgrade")));
+
+	nano::uint256_union node_id_mdb_key (3);
+	auto error (mdb_del (env.tx (transaction_a), meta, nano::mdb_val (node_id_mdb_key), nullptr));
+	release_assert (!error || error == MDB_NOTFOUND);
```
[the remark this answers is not in the document]
Comment (SergiySW) [reply nano/node/lmdb.cpp]: Good point

[the remark this answers is not in the document]
Comment (wezrule) [reply nano/node/lmdb.cpp]: Looks like there is a formatting issue reporter on travis. Otherwise LGTM

**Comment by SergiySW** [comment]
@cryptocode some projects also discover own peer IP with web services / other peers

**Comment by cryptocode** [comment]
@SergiySW yes indeed, I once suggested we'd do a curlmyip-ish implementation in the node - you'd query the preconfigured nodes to get the external ip. But because of various NAT setups, some projects allows you to set external port manually (in addition to external ip).

File: nano/node/node.cpp [hunk 2]
```diff
@@ -103,6 +103,22 @@ void nano::network::send_keepalive (nano::transport::channel const & channel_a)
...
+	nano::keepalive message;
+	udp_channels.random_fill (message.peers);
+	if (node.config.external_address != boost::asio::ip::address_v6{} && node.config.external_port != 0)
+	{
+		message.peers[0] = nano::endpoint (node.config.external_address, node.config.external_port);
+	}
+	else
+	{
+		message.peers[0] = nano::endpoint (boost::asio::ip::address_v6{}, endpoint ().port ());
+		message.peers[1] = nano::endpoint (boost::asio::ip::address_v6{}, node.port_mapping.external_address ().port ());
```
Comment (cryptocode) [inline nano/node/node.cpp]: UPnP also reports external ip. Do we wanna use that?

Comment (SergiySW) [reply nano/node/node.cpp]: We can use it as well, yes
To prevent possible upnp errors `message.peers[2] = node.port_mapping.external_address ();`

**Pull request merged**